docs/language-usage.rst

```diff
@@ -0,0 +1,354 @@
+:attr:`Document.HTML <gcloud.language.document.Document.HTML>`.
+
+In addition to supplying the text / HTML content, a document can refer
+to content stored in `Google Cloud Storage`_:
+
+  .. code-block:: python
+
+     >>> document = client.document(bucket='my-text-bucket',
+     ...                            blob='sentiment-me.txt')
+     >>> document.gcs_url
+     'gs://my-text-bucket/sentiment-me.txt'
+
+.. _analyzeEntities: https://cloud.google.com/natural-language/reference/rest/v1beta1/documents/analyzeEntities
+.. _analyzeSentiment: https://cloud.google.com/natural-language/reference/rest/v1beta1/documents/analyzeSentiment
+.. _annotateText: https://cloud.google.com/natural-language/reference/rest/v1beta1/documents/annotateText
+.. _Document: https://cloud.google.com/natural-language/reference/rest/v1beta1/Document
+.. _Google Cloud Storage: https://cloud.google.com/storage/
+
+Analyze Entities
+----------------
+
+The :meth:`~gcloud.language.document.Document.analyze_entities` method
+finds named entities (i.e. proper names) in the text, entity types,
+salience (prominence), mentions for each entity, and other properties.
+
+  .. code-block:: python
+
+     >>> text_val = ("Michelangelo Caravaggio, Italian painter, is "
+     ...             "known for 'The Calling of Saint Matthew'.")
+     >>> document = client.document(text_val)
+     >>> document.analyze_entities()
+     {
+       'entities': [
+         {
+           'name': 'Michelangelo Caravaggio',
+           'type': 'PERSON',
+           'metadata': {
+             'wikipedia_url': 'http://en.wikipedia.org/wiki/Caravaggio'
+           },
+           'salience': 0.75942981,
+           'mentions': [
+             {
+               'text': {
+                 'content': 'Michelangelo Caravaggio',
+                 'beginOffset': 0
+               }
+             }
+           ]
+         },
+         {
+           'name': 'Italian',
+           'type': 'LOCATION',
+           'metadata': {
+             'wikipedia_url': 'http://en.wikipedia.org/wiki/Italy'
+           },
+           'salience': 0.20193423,
+           'mentions': [
+             {
+               'text': {
+                 'content': 'Italian',
+                 'beginOffset': 25
+               }
+             }
+           ]
+         },
+         {
+           'name': 'The Calling of Saint Matthew',
+           'type': 'WORK_OF_ART',
+           'metadata': {
+             'wikipedia_url': 'http://en.wikipedia.org/wiki/index.html?curid=2838808'
+           },
+           'salience': 0.03863598,
+           'mentions': [
+             {
+               'text': {
+                 'content': 'The Calling of Saint Matthew',
+                 'beginOffset': 56
+               }
+             }
+           ]
+         }
+       ],
+       'language': 'en'
+     }
+
+In addition, the encoding of the text in the document can be
+specified in the request. By default, the UTF-8 is used:
```

Comment:
Right. The indexes aren't into the JSON-encoded utf8 byte string - they're into a string encoded in the specified encoding. A python unicode object is encoded in UTF32  (assuming the python interpreter was compiled in wide mode), which is how each code point is one index. So by specifying `encoding=UTF32`, the returned offsets will correspond to the python `unicode` object, which treats every code point as one index (rather than, say, UTF16, where triple-byte characters will occupy two indices). The reason I'm suggesting UTF32 be the default is _so that_ the user doesn't have to think about the encoding, since their string is probably already UTF32 (assuming they're using unicode objects, rather than python2 strings).

Then there's the issue that, depending on whether your python interpreter was compiled in [wide or narrow mode](http://stackoverflow.com/questions/13095499/unicode-in-python-just-utf-16), the unicode object might actually be UTF16.

Also note, if you're writing an app where you eventually pass the offsets to javascript (which encodes in UTF16), you'll need the offsets for UTF16 rather than UTF32. It all depends on what string you're going to be applying the offsets to. But if it's going to be applied to python `unicode` objects, it should be either UTF32 or UTF16, depending on `sys.maxunicode`